test/IECoreHoudini/SceneCacheTest.py

```diff
@@ -1432,6 +1432,67 @@ def testLiveTags( self ) :
 			self.assertTrue( scene.hasTag( tag, IECore.SceneInterface.EveryTag ) )
 		self.assertFalse( scene.hasTag( "notATag", IECore.SceneInterface.EveryTag ) )
 	
+	def testObjTagToGroups( self ) :
+		
+		self.writeDualTaggedSCC()
+		
+		xform = self.xform()
+
+		xform.parm( "hierarchy" ).set( IECoreHoudini.SceneCacheNode.Hierarchy.SubNetworks )
+		xform.parm( "depth" ).set( IECoreHoudini.SceneCacheNode.Depth.AllDescendants )
+		xform.parm( "tagFilter" ).set( "*" )
+		xform.parm( "expand" ).pressButton()
+
+		# no groups set by default
+		for src in ( c for c in xform.allSubChildren() if c.type().nameWithCategory() == "Sop/ieSceneCacheSource" ):
+			self.assertEqual( src.geometry().primGroups(), () )
+			
+		# all groups enabled
+		xform.parm( "tagGroups" ).set(1)
+		xform.parm( "tagGroupsPrefix" ).set("test_")
+		xform.parm( "push" ).pressButton()
+	
+		for src in (c for c in xform.allSubChildren() if c.type().nameWithCategory() == "Sop/ieSceneCacheSource"):
+			if src.path() == ( xform.path() + "/1/geo/1" ):
+				for gr in src.geometry().primGroups():
+					self.assertIn( gr.name(), [ "test_ObjectType_MeshPrimitive", "test_a" ] )
+			elif src.path() == ( xform.path() + "/1/2/geo/2" ):
+				for gr in src.geometry().primGroups():
+					self.assertIn( gr.name(), [ "test_ObjectType_MeshPrimitive", "test_b" ] )
+			elif src.path() == ( xform.path() + "/1/2/3/geo/3" ):
+				for gr in src.geometry( ).primGroups( ):
+					self.assertIn( gr.name( ), [ "test_ObjectType_MeshPrimitive", "test_c" ] )
+			elif src.path() == ( xform.path() + "/1/4/5/geo/5" ):
+				for gr in src.geometry( ).primGroups( ):
+					self.assertIn( gr.name( ), [ "test_ObjectType_MeshPrimitive" ] )
+
+		# group b filtered
+		xform.parm( "tagFilter" ).set( "b" )
+		xform.parm( "push" ).pressButton()
+
+		srcs = [ c for c in xform.allSubChildren() if c.type().nameWithCategory() == "Sop/ieSceneCacheSource" ]
+		self.assertEqual( set( [ pg.name() for src in srcs for pg in src.geometry().primGroups() ] ), {  "test_ObjectType_MeshPrimitive", "test_b" } )
+
+		# check that all groups present when Hierarchy is set to parenting
+		xform.parm( "collapse" ).pressButton()
+		xform.parm( "hierarchy" ).set( IECoreHoudini.SceneCacheNode.Hierarchy.Parenting )
+		xform.parm( "depth" ).set( IECoreHoudini.SceneCacheNode.Depth.AllDescendants )
+		xform.parm( "tagFilter" ).set( "*" )
+		xform.parm( "expand" ).pressButton()
+
+		srcs = [ c for c in xform.allSubChildren() if c.type().nameWithCategory() == "Sop/ieSceneCacheSource" ]
+		self.assertEqual( set( [ pg.name() for src in srcs for pg in src.geometry().primGroups() ] ), { "test_ObjectType_MeshPrimitive", "test_a", "test_b", "test_c" } )
```

Comment:
I think its worth testing explicitly here like you did above (e.g. test that each SOP has the specific tag we expect)